cli/azd/pkg/tools/dotnet/dotnet.go

```diff
@@ -192,6 +208,58 @@ func (cli *dotNetCli) PublishContainer(
 	return nil
 }
 
+func (cli *dotNetCli) GetTargetPort(
+	ctx context.Context, project, configuration string,
+) (int, error) {
+	runArgs := newDotNetRunArgs("publish", project,
```

Comment:
I think this will end up trying to also push the container into a local store or remote store.  We should pass [`ContainerArchiveOutputPath`](https://learn.microsoft.com/en-us/dotnet/core/docker/publish-as-container?pivots=dotnet-8-0#containerarchiveoutputpath) to a local temporary file which we delete after calling `dotnet publish`.

That should allow us to work even when docker is not installed locally and there is no host running to tag the image with.

Reply:
@hemarina and I were chatting about this today. I realized that the semantics of `msbuild /t:Target --getProperty:foobar` is that it executes the build target, then returns the evaluated output property. This means that `GetTargetPort` which translates to `dotnet publish /t:PublishContainer --getProperty:GeneratedContainerConfiguration` does a full ,NET publish before returning `GeneratedContainerConfiguration`. We wondered if we want to avoid this expensive operation, especially inside template evaluation.

We looked through the code a little, and wondered if a better approach is to generate a manifest that contains the following:
```yaml
ingress:
  targetPort: {{ .TargetPort }}
```

The `.TargetPort` expression is only evaluated briefly during the `deploy` command, after `dotnet publish` while gathering results, and before submitting the manifest. Maybe this is more in line with your other comment about delayed evaluation, @ellismg. I claim no credit to this as this was @hemarina's original approach..